netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,169 @@
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.filter(t => t.text.equalsIgnoreCase(token.text) &&
+      (editorArea.offsetToLine(editorArea.getDocument.asInstanceOf[PlainDocument],t.start)) !=
+        (editorArea.offsetToLine(editorArea.getDocument.asInstanceOf[PlainDocument], token.start)))
+  }
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+
+  class LineRenderer(boldedString: Option[String]) extends TableCellRenderer {
+    override def getTableCellRendererComponent(table: JTable, value: AnyRef, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component = {
+      val pane  = new JEditorPane()
+      pane.setOpaque(true)
+      pane.setBorder(BorderFactory.createEmptyBorder(1,0,0,0))
+      val editorKit = boldedString match {
+        case None => new HighlightEditorKit(null, editorArea.colorizer)
+        case Some(selectedString) => new BoldEditorKit(selectedString)
+      }
+      pane.setEditorKit(editorKit)
+      pane.setText(value.asInstanceOf[String])
+      val alternate = UIManager.getColor("Table.alternateRowColor")
+      val defaults = new UIDefaults()
+      if (row%2 == 0){
+        defaults.put("EditorPane[Enabled].backgroundPainter", Color.WHITE)
+        pane.setBackground(Color.WHITE)
+      }
+      else {
+        defaults.put("EditorPane[Enabled].backgroundPainter", alternate)
+        pane.setBackground(alternate)
+      }
+      if(isSelected) {
+        pane.setBackground(usageTable.getSelectionBackground)
+        pane.setForeground(usageTable.getSelectionForeground)
+      }
+      pane.putClientProperty("Nimbus.Overrides", defaults)
+      pane.putClientProperty("Nimbus.Overrides.InheritDefaults", true)
+      pane
+    }
+  }
+  class LineNumberRenderer extends DefaultTableCellRenderer {
+    override def setValue(value: AnyRef) = {
+      setText(editorArea.offsetToLine(editorArea.getDocument.asInstanceOf[PlainDocument], value.asInstanceOf[Token].start).toString)
+    }
+  }
+  import org.nlogo.editor.{ HighlightEditorKit, HighlightView }
+  class BoldEditorKit(selectedString: String) extends HighlightEditorKit(null, editorArea.colorizer) {
+    import scala.collection.immutable.Range
+    class BoldView(elem: javax.swing.text.Element) extends HighlightView(this.pane, elem, this.colorizer) {
+      var boldingRange = Option.empty[Range]
+      override def studyLine(lineIndex: Int): Unit = {
+        super.studyLine(lineIndex)
+        val elem = getElement.getElement(lineIndex)
+        val lineText = getDocument.getText(elem.getStartOffset, elem.getEndOffset - elem.getStartOffset max 0)
+        val beginningIndex = lineText.toUpperCase.indexOf(selectedString.toUpperCase)
```

Comment:
Sounds good. I wrote this code knowing there were some important cases it wouldn't work exactly properly on for the purpose of illustrating how to bold certain parts of text.